File: 滑动窗口/3.无重复的最长子串.go
package 滑动窗口

func lengthOfLongestSubstring(s string) int {
	n := len(s)
	cnt := make([]int, 128)
	l, res := 0, 0
	for r := 0; r < n; r++ {
		for cnt[s[r]] > 0 {
			cnt[s[l]]--
			l++
		}
		cnt[s[r]] = 1
		res = max(res, r-l+1)
	}
	return res
}

File: 滑动窗口/3.无重复的最长子串.java
class Solution {
  public int lengthOfLongestSubstring(String s) {
    char[] strs = s.toCharArray();
    Map<Character,Integer>map = new HashMap<>();
    int l=0,r=0,res=0;
    for(;r<strs.length;r++){
      while(map.containsKey(strs[r])){
        map.remove(strs[l]);
        l++;
      }
      map.put(strs[r],1);
      res=Math.max(res,r-l+1);
    }
    return res;

  }
}

